proto/asn1-pdu/x509_certificate_to_der.cc

```diff
@@ -43,6 +43,172 @@ DECLARE_ENCODE_FUNCTION(AlgorithmIdentifierSequence) {
   EncodeTagAndLength(kAsn1Sequence, der.size() - tag_len_pos, tag_len_pos, der);
 }
 
+DECLARE_ENCODE_FUNCTION(ExtendedKeyUsage) {
+  Encode(val.key_purpose_id(), der);
+  for (const auto& key_purpose_id : val.key_purpose_ids()) {
+    Encode(key_purpose_id, der);
+  }
+}
+
+DECLARE_ENCODE_FUNCTION(BasicConstraints) {
+  if (val.ca().val()) {
```

Comment:
Maybe add a note here that it's omitted for default value

```suggestion
  // RFC 5280, 4.2.1.9
  //    BasicConstraints ::= SEQUENCE {
  //        cA                      BOOLEAN DEFAULT FALSE,
  //        pathLenConstraint       INTEGER (0..MAX) OPTIONAL }
  //
  //  Omit basicConstraints if it's false, as that's a DEFAULT value in a SEQUENCE (see
  // X.690 (2015), 11.5)
  if (val.ca().val()) {
```